pontoon/base/models.py

```diff
@@ -89,17 +90,40 @@ def translations_count(query=None):
         )
 
 
-# User class extensions
+@property
+def user_translated_locales(self):
+    locales = []
+
+    for group in self.groups.all():
+        if hasattr(group, 'translated_locales'):
+            locale = group.translated_locales.first()
+            if locale:
+                locales.append(locale)
+        if hasattr(group, 'managed_locales'):
+            locale = group.managed_locales.first()
+            if locale:
+                locales.append(locale)
+
+    return [l.code for l in set(locales)]
+
+
+@property
+def user_display_permissions(self):
```

Comment:
A macro would be okay but I prefer helpers for things like this because it's mostly logic rather than markup. If it was more markup-heavy than a macro would make more sense.

That's my preference though, it doesn't make a huge difference either way.